src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,42 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+
+/**
+ * Mimics [com.intellij.formatting.alignment.AlignmentStrategy], but offers more flexible API
+ */
+interface RustAlignmentStrategy {
```

Comment:
>  Will [this](https://github.com/intellij-rust/intellij-rust/pull/413#discussion_r64387669) help?

Hmm I'm tinkering around it now and it turns out that formatting blocks are created BFS-style: block's grandchildren are created **after** it's children are fully created. This means that in order to pass alignment object we have to:
  - store it somewhere for the whole lifetime of fn decl block (property or user data)...
  - ...or do some reference-passing shenaningans (this means introducing lots of separate formatting blocks)

OMG this is going to yield horrible code :sob: Neither of these options is appealing to me.